foundations/foundation-metrics/src/main/java/io/servicecomb/foundation/metrics/event/MetricsEventManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Huawei Technologies Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.foundation.metrics.event;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class MetricsEventManager {
+  private static final Map<Class<? extends MetricsEvent>, List<MetricsEventListener>> allEvents = new ConcurrentHashMap<>();
+
+  public static void registerEventListener(MetricsEventListener eventListener) {
```

Comment:
ConcurrentHashMap means we need to support concurrent register
but arrayList.add is not safe.

and registerEventListener will be invoked rarely
so after registerEventListener is safe, maybe need to  change from arrayList to CopyOnWrite list to make triggerEvent safe

Reply:
Done